src/posix.h

```diff
@@ -0,0 +1,45 @@
+#ifndef BENCHMARK_POSIX_H_
+#define BENCHMARK_POSIX_H_
+
+#include <cstdio>  // fileno, _fileno
+#include <cstdlib> // getenv
+
+#include "internal_macros.h"
+
+#ifdef BENCHMARK_OS_WINDOWS
+#include <io.h> // _isatty
+#else
+#include <unistd.h> // isatty
+#endif
+
+namespace benchmark {
+
+inline const char* GetEnv(const char* name) {
+#if defined(__BORLANDC__) || defined(__SunOS_5_8) || defined(__SunOS_5_9)
+  // Environment variables which we programmatically clear will be set to the
+  // empty string rather than unset (nullptr).  Handle that case.
+  const char* const env = getenv(name);
+  return (env != nullptr && env[0] != '\0') ? env : nullptr;
+#else
+  return getenv(name);
+#endif // BENCHMARK_OS_WINDOWS
+}
+
+inline int IsATTY(int fd) {
```

Comment:
Actually since `IsATTY` and `FileNo` seem like unneeded abstractions since they only have one caller, `IsColorTerminal`. I would remove these and just call `isatty` and `fileno` directly from there.